File: src/test/java/com/github/kayjamlang/tests/NullableTypeTest.java
package com.github.kayjamlang.tests;

import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.Type;
import com.github.kayjamlang.core.exceptions.LexerException;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class NullableTypeTest {
    private static KayJamParser parser;

    @BeforeClass
    public static void prepare() {
        parser = new KayJamParser(new KayJamLexer("string?"));
    }

    @Test
    public void test() throws Exception {
        Type type = parser.parseType(false);
        assertEquals(Type.STRING, type);
        assertTrue(type.nullable);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/NullableTypeTest.scala
package com.github.kayjamlang.tests

import com.github.kayjamlang.core.KayJamLexer
import com.github.kayjamlang.core.KayJamParser
import com.github.kayjamlang.core.Type
import org.junit.BeforeClass
import org.junit.Test
import org.junit.Assert._


object NullableTypeTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("string?"))
  }
}

class NullableTypeTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val `type` = NullableTypeTest.parser.parseType(false)
    assertEquals(Type.STRING, `type`)
    assertTrue(`type`.nullable)
  }
}
